src/AI/WallyAI.java

```diff
@@ -815,9 +918,9 @@ private static int valueUnit(Unit unit, Location locale, boolean includeCurrentH
   {
     int value = unit.model.getCost();
 
-    if (includeCurrentHealth)
+    if( includeCurrentHealth )
       value *= unit.getHP();
-    value *= (unit.getCaptureProgress() > 0 && locale.isCaptureable())? 8 : 1; // Strongly value capturing units
+    value *= (unit.model.hasActionType(UnitActionFactory.CAPTURE) && locale.isCaptureable()) ? 8 : 1; // Strongly value capturing units
```

Comment:
Isn't capture progress a more true metric of whether a unit is actually capturing? Checking the action type tells whether it _could_ start capturing, but then I might argue you should also check if there are any capturable tiles in range, not just directly underneath.

Reply:
True, but this means you could e.g. kick a Tank off of a city so an inf can capture it, or kick a lower-HP inf off so you can capture it (if it hadn't used its turn capturing already).
It also means Wally's very interested in getting enemy troops off *his* property.

It *does* provide the "interesting" property that Wally values infantry on properties more even on their own turf, so I guess I should fix that.